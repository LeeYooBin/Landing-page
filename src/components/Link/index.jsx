import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Content = styled.a`
  text-decoration: none;
  color: ${({ theme }) => theme.colors.text};
  font: ${({ theme }) => theme.fonts.text1};
  position: relative;

  @media (min-width: 920px) {
    text-align: center;
    color: ${({ theme }) => theme.colors.text};
    margin-left: 5vw;

    &:first-child {
      margin: 0;
    }

    &::after {
      content: '';
      position: absolute;
      bottom: 0;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${({ theme }) => theme.colors.sun};
      transition: all 300ms ease-in-out;
    }

    &:hover::after {
      left: 0;
      width: 100%;
    }
  }
`;

const Link = ({ link = '' }) => (
  <Content href={`#${link}`}>
    {link.replace(/^\w/, (match) => match.toUpperCase())}
  </Content>
);

Link.propTypes = {
  link: PropTypes.string,
};

export default Link;
