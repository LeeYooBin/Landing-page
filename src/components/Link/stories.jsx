import Link from '.';

export default {
  title: 'Link',
  component: Link,
};

export const Template = (args) => <Link {...args} />;
