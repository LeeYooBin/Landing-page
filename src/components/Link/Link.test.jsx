/* eslint-disable jsx-a11y/anchor-is-valid */
import { describe, it, expect } from '@jest/globals';
import { screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Link from '.';
import renderTheme from '../../styles/render-theme';

describe('Text component', () => {
  it('should render with default values', () => {
    renderTheme(<Link />);
  });

  it('renders Link component', () => {
    renderTheme(<Link link="example" />);
    expect(screen.getByText('Example')).toBeInTheDocument();
  });

  it('generates link correctly', () => {
    renderTheme(<Link link="example" />);
    const linkElement = screen.getByText('Example');
    expect(linkElement.getAttribute('href')).toBe('#example');
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(<Link />);
    expect(container).toMatchSnapshot();
  });
});
