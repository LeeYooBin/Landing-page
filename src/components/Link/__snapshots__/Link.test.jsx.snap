// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Should match snapshot 1`] = `
.c0 {
  text-decoration: none;
  color: #FFF;
  font: 400 1.6rem/2.4rem "Heebo",sans-serif;
  position: relative;
}

@media (min-width: 920px) {
  .c0 {
    text-align: center;
    color: #FFF;
    margin-left: 5vw;
  }

  .c0:first-child {
    margin: 0;
  }

  .c0::after {
    content: '';
    position: absolute;
    bottom: 0;
    left: 50%;
    width: 0;
    height: 0.2rem;
    background: #F5D15F;
    transition: all 300ms ease-in-out;
  }

  .c0:hover::after {
    left: 0;
    width: 100%;
  }
}

<div>
  <a
    class="c0"
    href="#"
  />
</div>
`;
